src/bun.js/api/bun/h2_frame_parser.zig

```diff
@@ -0,0 +1,2563 @@
+                        return JSC.JSValue.jsNumber(stream.id);
+                    };
+                }
+            }
+        }
+
+        var flags: u8 = @intFromEnum(HeadersFrameFlags.END_HEADERS);
+        var exclusive: bool = false;
+        var has_priority: bool = false;
+        var weight: i32 = 0;
+        var parent: i32 = 0;
+        var waitForTrailers: bool = false;
+        var end_stream: bool = false;
+        if (args_list.len > 2 and !args_list.ptr[2].isEmptyOrUndefinedOrNull()) {
+            const options = args_list.ptr[2];
+            if (!options.isObject()) {
+                stream.state = .CLOSED;
+                stream.rstCode = @intFromEnum(ErrorCode.INTERNAL_ERROR);
+                this.dispatchWithExtra(.onStreamError, JSC.JSValue.jsNumber(stream_id), JSC.JSValue.jsNumber(stream.rstCode));
+                return JSC.JSValue.jsNumber(stream.id);
+            }
+
+            if (options.get(globalObject, "waitForTrailers")) |trailes_js| {
+                if (trailes_js.isBoolean()) {
+                    waitForTrailers = trailes_js.asBoolean();
+                    stream.waitForTrailers = waitForTrailers;
+                }
+            }
+
+            if (options.get(globalObject, "endStream")) |end_stream_js| {
+                if (end_stream_js.isBoolean()) {
+                    if (end_stream_js.asBoolean()) {
+                        end_stream = true;
+                        // will end the stream after trailers
+                        if (!waitForTrailers) {
+                            flags |= @intFromEnum(HeadersFrameFlags.END_STREAM);
+                        }
+                    }
+                }
+            }
+
+            if (options.get(globalObject, "exclusive")) |exclusive_js| {
+                if (exclusive_js.isBoolean()) {
+                    if (exclusive_js.asBoolean()) {
+                        exclusive = true;
+                        stream.exclusive = true;
+                        has_priority = true;
+                    }
+                }
+            }
+
+            if (options.get(globalObject, "parent")) |parent_js| {
+                if (parent_js.isNumber() or parent_js.isInt32()) {
+                    has_priority = true;
+                    parent = parent_js.toInt32();
+                    if (parent <= 0 or parent > MAX_STREAM_ID) {
+                        stream.state = .CLOSED;
+                        stream.rstCode = @intFromEnum(ErrorCode.INTERNAL_ERROR);
+                        this.dispatchWithExtra(.onStreamError, JSC.JSValue.jsNumber(stream_id), JSC.JSValue.jsNumber(stream.rstCode));
+                        return JSC.JSValue.jsNumber(stream.id);
+                    }
+                    stream.streamDependency = @intCast(parent);
+                }
+            }
+
+            if (options.get(globalObject, "weight")) |weight_js| {
+                if (weight_js.isNumber() or weight_js.isInt32()) {
```

Comment:
if `weight` is `NaN` what happens?

Reply:
hmm will check, to see what happens, we have some isNaN() on JSValue?